src/plots/cartesian/graph_interact.js

```diff
@@ -624,10 +624,14 @@ function hover(gd, evt, subplot) {
     if(!hoverChanged(gd, evt, oldhoverdata)) return;
 
     if(oldhoverdata) {
-        gd.emit('plotly_unhover', { points: oldhoverdata });
+        gd.emit('plotly_unhover', {
+            event: evt,
+            points: oldhoverdata
+        });
     }
 
     gd.emit('plotly_hover', {
+        event: evt,
```

Comment:
@etpinard It doesn't work for pie, ternary, geo and mapbox, because when they call to `Fx.click(gd, evt)` they make up an `evt`.

I reckon that to make it work for all these types of subplots, first I need to locate where they call `Fx.click`, `Fx.hover` and `Fx.unhover` and  then see if it's possible to send the original event.

For example, [here](https://github.com/n-riesco/plotly.js/blob/aea866efd5e93765e6f1fc876b11fa7d941e1dde/src/traces/pie/plot.js#L148): why do we need to set the event to `{target: true}`? Is is OK if we just do `Fx.click(gd, d3.event);`